### DISCUSSION TITLE
Errors inside map function

### DISCUSSION THREAD

**Discussion by emil-malmgaard-rasmussen** [react-hook-form/react-hook-form#5476] [Q&A] [answered]
Hi everyone

I'm relative new to react-hook-form, so it might be me doing it the wrong way, but I have a problem trying to show errors from a map function, what I've come to is:

```
const {handleSubmit, control, register, errors} = useForm();

...
{genericDrawerState.data.questionLines.map((category, index) => (
<input name={`questionLine[${index}]`} type="text" ref={register({required: true})} defaultValue={category.title} />
 {errors.questionLine[index] && errorMessage}
)).reverse()}
...
```
As mentioned above, I'm trying to loop over and array of objects, if the default value is deleted and they try to submit the input to the form, it should dispaly a custom error message.

I am getting an error saying errors.questionLine[0] doesn't exist, but my impression was, that the name={} in input, registers inside the useForm? 

This might be a no-brainer for some of you folks, sorry for the newbie question

Thank you very much for the library, really helped building my application!

**Comment by emil-malmgaard-rasmussen** [comment, marked as answer]
anyone?

**Reply by bluebill1049** [reply]
can you provide a codesandbox on the issue above?

**Reply by emil-malmgaard-rasmussen** [reply]
Thanks for taking your time @bluebill1049 ! :) 

I've tried making a scraped version here:
https://codesandbox.io/s/solitary-sea-lxhg8
What im trying to accomplish is to get an error message for each input field: 
``` 
{errors.inputField[index] && errorMessage}
```
But that is returning undefined, I'm pretty sure that it's me doing something wrong, so i'm really just curious as how one would accomplish this.

If i do {errors.inputField[0] && errorMessage} it works, but only for inputField[0] 

Thanks a lot again!

**Reply by emil-malmgaard-rasmussen** [reply]
is the above provided sandbox sufficient? @bluebill1049 🙂

**Reply by bluebill1049** [reply]
`{errors.inputField && errors.inputField[index] && errorMessage}`

**Reply by emil-malmgaard-rasmussen** [reply]
Hi @bluebill1049
Sorry for the late reply! 

It worked like a charm, thank you for this amazing library and the support you provide!
